実装機能1：Home画面にお知らせ一覧表示 (PORTAL-37)　実装機能2：お知らせ登録SQL文の修正 (PORTAL-38)

BSS-OnumaTakuto opened the pull request:
# Jira Ticket
​1．お知らせタイトルを表示
https://bss-portal.atlassian.net/browse/PORTAL-37
2．お知らせ登録SQL文の修正
https://bss-portal.atlassian.net/browse/PORTAL-38
​

# やった事
​1．お知らせタイトルを表示
​Home画面にお知らせ一覧が表示されるように修正しました。
また、ログイン後の画面遷移でお知らせ一覧が取得できない問題がありましたが、LoginController.javaを修正することで取得できない問題も解決しました。

2．お知らせ登録SQL文の修正
DefaultのSQL文だと、掲載期間_開始 「start_time」、掲載期間_終了 「end_time」にNotNull制約が記載されていますが、
全てに掲載期間を設定する必要はなく、掲載期間が必要なものだけ期間を設定すれば良いと考えたため、NotNull制約の記述を削除しました。
​
3．プロジェクト定義書＿お知らせ登録.xlsx の修正
掲載期間_開始 「start_time」、掲載期間_終了 「end_time」のNotNull制約を削除しました。


# なぜやるのか
1．お知らせタイトルを表示
​Home画面に表示されていたお知らせ内容がhtmlファイルに直書きされていたため。

2．お知らせ登録SQL文の修正
NotNull制約があると掲載期間を設けたくない内容にも期間を設ける必要が発生するため。

​
# 動作確認
​windowsのEclipseで実行し、挙動確認を行いました。

【Home画面】
<img width="960" alt="スクリーンショット 2023-10-19 100046" src="https://github.com/miaochshBss/ShainKintaiKanriBss/assets/147013917/284c674e-d56b-4ee0-9059-02dbd3c99082">
​
【Notifications基本情報画面】プロジェクト定義書＿お知らせ登録.xlsx 
<img width="960" alt="スクリーンショット 2023-10-19 104344" src="https://github.com/miaochshBss/ShainKintaiKanriBss/assets/147013917/7bd296c9-1bc0-490d-be50-28eeea9c876d">


# Refs (レビューにあたって参考にすべき情報）(Optional)
​Home画面のお知らせデータはタイトルの取得が主な目的だと考えたため、変数名は「notificationTitle」としています。

レビュー、よろしくお願いします。

BSS-OnumaTakuto committed:
実装機能：Home画面にお知らせ一覧表示

Home画面にお知らせ一覧表示

BSS-OnumaTakuto committed:
SQL修正：notifications.sqlの not null制約削除

BSS-OnumaTakuto committed:
home.htmlが抜けていました。

BSS-OnumaTakuto committed:
Merge branch 'feature-dev-05-PORTAL-37' into feature-dev-05

bss-nagasawa commented on SQL/notifications.sql:
create table public.notifications (
  id serial not null
  , title character varying(255) not null
  , content text not null
  , status character varying(50) default '通常'
  , file_content bytea
  , start_time timestamp(6) without time zone not null
  , end_time timestamp(6) without time zone not null
  , posted_by integer
  , created_at timestamp(6) without time zone default CURRENT_TIMESTAMP
  , updated_at timestamp(6) without time zone default CURRENT_TIMESTAMP
  , creation_date timestamp(6) without time zone default CURRENT_TIMESTAMP
  , creator integer
  , update_date timestamp(6) without time zone default CURRENT_TIMESTAMP
  , updater integer
  , primary key (id)
);
全体を見たとき、
posted_byとcreatorは同意、
created_atとcreation_dateは同意、
になると思うのでどちらかに寄せてください。
日付は〇〇_dateに寄せたほうがわかりやすいです。

bss-nagasawa replied:
カラム名の修正をお願いします。
id , title , content , status , file_content , start_time , end_time , creater , creation_date , updater , update_date

BSS-OnumaTakuto replied:
下記の内容で **notifications.sql** のカラム名を修正いたしました。
id , title , content , status , file_content , start_time , end_time , creater , creation_date , updater , update_date

また、statusのカラムの型を varchar(1) に変更し、「0:重要 1:通常」デフォルト 1 としました。
併せて「**プロジェクト定義書＿お知らせ登録.xlsx**」も上記の内容で修正しました。

【Notifications基本情報画面】プロジェクト定義書＿お知らせ登録.xlsx
<img width="960" alt="スクリーンショット 2023-10-19 155243" src="https://github.com/miaochshBss/ShainKintaiKanriBss/assets/147013917/7e69c785-3f6d-4655-a88f-6e30b9d71ffb">

ご確認の程、よろしくお願いいたします。

BSS-OnumaTakuto committed:
notifications.sqlのカラム変更

お知らせ登録のカラムを下記の内容に修正しました。
id , title , content , status , file_content , start_time , end_time , creater , creation_date , updater , update_date

それに伴い、プロジェクト定義書＿お知らせ登録.xlsxの内容も同様に修正しました。

BSS-OnumaTakuto committed:
Merge branch 'feature-dev-05-POETAL-38' into feature-dev-05
